### PR TITLE
Small fixes for upcoming release

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**

After checking new features, some minor fixes are needed.

### CODE REVIEW

File: pennylane/templates/subroutines/interferometer.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def circuit(params):
 
         Using these random parameters, the resulting circuit is:
 
-        >>> print(qml.draw(circuit)(params))
+        >>> print(qml.draw(circuit, expansion_strategy="device")(params))
```
Comment (antalszava) [inline pennylane/templates/subroutines/interferometer.py:158]: What was the change in the codebase that caused this to be required?

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -110,7 +110,7 @@
   ...     (x, Y), history = opt.step(qnode, x, Y, spectra=spectra, full_output=True)
   ...     print(f"New cost: {np.round(qnode(x, Y), 3)} reached via substeps {np.round(history, 3)}")
   New cost: 0.0 reached via substeps [-0.  0.  0.]
-  New cost: -1.0 reached via substeps [-0.276 -0.276 -1.   ]
+  New cost: -1.0 reached via substeps [-1. -1. -1.]
```
Comment (antalszava) [inline doc/releases/changelog-dev.md:113]: How come this changed?

**Review by antalszava — APPROVED** [review]
Looks good :100: Just curious about the reasons for the changes.

File: pennylane/templates/subroutines/interferometer.py (hunk 1 above)
Comment (rmoyard) [reply pennylane/templates/subroutines/interferometer.py:158]: I think this is due to the refractor of the Interferometer template https://github.com/PennyLaneAI/pennylane/pull/1946/files

File: doc/releases/changelog-dev.md (hunk 2 above)
Comment (rmoyard) [reply doc/releases/changelog-dev.md:113]: I run the whole example and had different values only for this array.

Comment (antalszava) [reply doc/releases/changelog-dev.md:113]: I see, but why is that?

Comment (rmoyard) [reply doc/releases/changelog-dev.md:113]: It was probably a copy paste typo

**Pull request merged**